crates/ghost_actor/tests/manual_example_mod/test_actor.rs

```diff
@@ -5,12 +5,23 @@
 use super::test_protocol::*;
 use ghost_actor::prelude::*;
 
+pub type GhostSystemRef<'lt> = SingleThreadedGhostSystemRef<'lt>;
+
 pub struct TestActor<'lt> {
-    owner_ref: GhostEndpointFull<'lt, TestProtocol, (), Self, TestActorHandler<'lt, Self>>,
+    owner_ref: GhostEndpointFull<
+        'lt,
+        TestProtocol,
+        (),
+        Self,
+        TestActorHandler<'lt, Self>,
+        GhostSystemRef<'lt>,
```

Comment:
Also here, we probably want our actors be be able to live in multiple systems, so we should make the system be a generic.